xdsl/transforms/transform_interpreter.py

```diff
@@ -0,0 +1,43 @@
+"""Transform interpreter."""
+
+from dataclasses import dataclass
+
+from xdsl.dialects import builtin, transform
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.transform import TransformFunctions
+from xdsl.passes import Context, ModulePass
+from xdsl.utils.exceptions import PassFailedException
+
+
+@dataclass(frozen=True)
+class TransformInterpreterPass(ModulePass):
+    """Transform dialect interpreter"""
+
+    # TODO: Add the rest of options to the transform interpreter
+    # and add semantics for all of the other transform operations
+
+    name = "transform-interpreter"
+
+    entry_point: str = "__transform_main"
```

Comment:
if we provide a way to manually specify the entry point, we should add tests that leverage it, as well as a test for the case where the entry point is not found

Reply:
I cannot test with filecheck with expected-error@below. But I added a test with another entry point and a pytest for the exception.